Database/Buffer/Backend/index.ts

```diff
@@ -1,13 +1,104 @@
+import * as isoly from "isoly"
 import "./load"
 import "./store"
+import { Archive } from "../../../Database/Archive"
+import { Configuration } from "../../../Database/Configuration"
+import { Document } from "../../../Database/Document"
+import { KeyValueStore } from "../../../KeyValueStore"
 import { DurableObjectState } from "../../../platform"
+import { Key } from "../../Key"
 import { Context } from "./Context"
 import { Environment } from "./Environment"
+import { Storage } from "./Storage"
 
 export class Backend {
-	private constructor(private readonly state: DurableObjectState) {}
+	alarmTime = 10 * 1000
+	partitions?: string[]
+	idLength?: number
+	private constructor(private readonly state: DurableObjectState, private environment: Environment) {
+		this.state.blockConcurrencyWhile(async () => {
+			this.partitions = await this.state.storage.get("partitions")
+			!!(await this.state.storage.getAlarm()) ||
+				(await (async () => {
+					await this.state.storage.setAlarm(this.alarmTime)
+					return true
+				})())
+		})
+	}
+
+	async fetch(request: Request): Promise<Response> {
+		this.state.waitUntil(this.configuration(request))
+		return await Context.handle(request, { ...(this.environment ?? {}), state: this.state })
+	}
+
+	async configuration(request: Request): Promise<void> {
+		const idLength = +(request.headers.get("length") ?? NaN)
+		this.idLength = Number.isNaN(idLength) ? undefined : idLength
+		const partitions = request.headers.get("partitions")?.split("/").slice(0, -1)
+		!this.partitions &&
+			!(this.partitions = await this.state.storage.get("partitions")) &&
+			partitions &&
+			this.state.waitUntil(this.state.storage.put("partitions", (this.partitions = partitions)))
+	}
+
+	async alarm(): Promise<void> {
+		const archiveTime = 12 * 1000 //after settlement
+		const archiveThreshold = isoly.DateTime.create(Date.now() - archiveTime, "milliseconds").substring(0, 19) //replace by new isoly function
```

Comment:
There are two functions that can do this.
Either
```ts
isoly.DateTime.previous(dateTime, {milliseconds})
```
or 
```ts
isoly.DateTime.previousMilliseconds(dateTime, milliseconds)
```
Altought you would still have to to `substring(0,19)`